lib/_tls_wrap.js

```diff
@@ -582,6 +593,22 @@ function TLSSocket(socket, opts) {
   if (enableTrace && this._handle)
     this._handle.enableTrace();
 
+  if (wrapHasActiveWriteFromPrevOwner) {
+    // `wrap` is a streams.Writable in JS. This empty write will be queued
+    // and hence finish after all existing writes, which is the timing
+    // we want to start to send any tls data to `wrap`.
+    const that = this;
```

Comment:
This shouldn't be needed for arrow functions.

Reply:
Got it. Thx :)